app/assets/javascripts/Components/Modals/graders_distribution_modal.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import Modal from "react-modal";
+import PropTypes from "prop-types";
+
+export class GraderDistributionModal extends React.Component {
+  static defaultProps = {
+    override: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.input = React.createRef();
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+  }
+
+  onSubmit = event => {
+    event.preventDefault();
+    const weightings = {};
+    const form = new FormData(this.input.current);
+    form.forEach((key, value) => {
+      weightings[value] = key;
+    });
+    this.props.onSubmit(weightings);
+  };
+
+  renderGraderRow = grader => {
+    return (
+      <div className="flex-row-expand" key={grader.user_name}>
+        <label htmlFor={`input-${grader.user_name}`} className="modal-inline-label">
+          {grader.user_name}
+        </label>
+        <input
+          id={`input-${grader.user_name}`}
+          type="number"
+          step="0.01"
+          min="0"
+          max="100"
+          defaultValue="1"
+          name={`${grader._id}`}
+          required={true}
+        />
+      </div>
+    );
+  };
+
+  render() {
+    return (
+      <Modal
+        className="react-modal dialog"
+        isOpen={this.props.isOpen}
+        onRequestClose={this.props.onRequestClose}
+      >
+        <form id="grader-form-random" onSubmit={this.onSubmit} ref={this.input}>
+          <div className={"modal-container-vertical"}>
+            <h2>{I18n.t("graders.weight")}</h2>
+            <p className="word-wrap">{I18n.t("graders.random_instruction")}</p>
+            <a href="https://github.com/MarkUsProject/Wiki/blob/release/Instructor-Guide--Assignments--Assigning-Graders.md">
```

Comment:
This isn't exactly quite right, as we support linking to a specific version of the documentation. In other places in the code this is done with the `@markus_version` instance variable, however this isn't available in the pure javascript code.

The way we've handled this in other situations is to add a line like `const MARKUS_VERSION = <%= @markus_version %>;` in the view layouts, creating a global constant that you can access in Javascript code. Check out how we do this for `AUTH_TOKEN`.